### ISSUE TITLE
Uploading media using auth context

### ISSUE THREAD

**Issue by botherder** [dghubble/go-twitter#121]
Walking through the documentation it doesn't appear immediately possible to use go-twitter to upload media files to attach to status updates. More precisely I'm referring to the ability to use this:
https://developer.twitter.com/en/docs/media/upload-media/api-reference/post-media-upload.html

Is there any particular way you recommend re-using go-twitter authentication procedure to authenticate to upload these media files as well?

Thanks

**Comment by botherder** [comment]
I resolved re-using the HTTP client.

**Comment by lexiross** [comment]
@botherder - would love to know how you resolved this issue! Currently having some trouble re-using the http client for media upload requests. Did you do anything special to handle auth? (see #133)

**Comment by botherder** [comment]
@lexiross First I set up the various Twitter and HTTP clients:

```
config = oauth1.NewConfig(consumerKey, consumerSecret)
token = oauth1.NewToken(accessToken, accessSecret)
httpClient = config.Client(oauth1.NoContext, token)
client = twitter.NewClient(httpClient)
```

Then I create a new Request `http.NewRequest` to `https://upload.twitter.com/1.1/media/upload.json`:

```req, _ := http.NewRequest("POST", "https://upload.twitter.com/1.1/media/upload.json", strings.NewReader(data.Encode()))```

Then I reuse `httpClient` to perform that request:

```resp, err := httpClient.Do(req)```

**Comment by lexiross** [comment]
@botherder thanks so much for your help! When I copied your code (with the encoded request body = `{"command": "INIT", "media_category":"tweet_gif","media_type":"image/gif","total_bytes":3000000}`) I get the following error from Twitter: `{"code":38,"message":"media parameter is missing."}`

Looks like other folks who've had that problem weren't setting the Content-Type header properly, so I tried setting it to `application/x-www-form-urlencoded` per the docs, but now I'm back to getting 401s. Did you encode just the init parameters into the request body as I did, or did you combine the `append` and `finalize` params as well?
